### PR TITLE
[NodeBalancer Refactor] - Request Config Nodes on Edit Landing Mount

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Creates new Redux State shape for NodeBalancer Config Nodes:

```js
/** each ID here is a Config ID */
{
 123: {
      data: ConfigNode[],
      loading: false,
      error: {}
  },
  234: {
      data: ConfigNode[] ,
      loading: false,
      error: {}
  }
}
```

This also requests the nodebalancer config nodes when the configuration page is mounted

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts [hunk 1]
```diff
@@ -1,90 +1,87 @@
...
-) => ReduxStateProps & Partial<StateProps>;
+  data: State
+) => ReduxStateProps & Partial<State>;
 
-export type Props = DispatchProps & StateProps;
+export type Props = DispatchProps & State;
 
 interface Connected {
   <ReduxStateProps, OwnProps>(
+    configID: number,
```
Comment (johnwcallahan) [inline packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts]: How will this work when it's connected to a component? We might need to get it from `ownProps`.

File: packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts [hunk 2]
```diff
@@ -1,90 +1,87 @@
...
-        nodeBalancerConfigNodesLastUpdated: lastUpdated,
-        nodeBalancerConfigNodesLoading: loading,
-        nodeBalancerConfigNodesResults: results.length
-      };
+      if (mapStateToProps) {
+        return mapStateToProps(ownProps, configNodesForAConfig);
+      }
 
-      return mapStateToProps ? mapStateToProps(ownProps, result) : result;
+      return configNodesForAConfig;
```
Comment (johnwcallahan) [inline packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts]: I think this should be `{ data: configNodesForAConfig }`.

File: packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts [hunk 3]
```diff
@@ -1,90 +1,87 @@
...
     (state, ownProps) => {
-      const {
-        loading,
-        error,
-        entities,
-        results,
-        lastUpdated
-      } = state.__resources.nodeBalancerConfigNodes;
+      const { nodeBalancerConfigNodes } = state.__resources;
+      const configNodesForAConfig = Object.keys(nodeBalancerConfigNodes).reduce(
```
Comment (martinmckenna) [inline packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts]: will change to `.values`. I don't know why I can't remember this

File: packages/manager/src/store/nodeBalancerConfigNodes/configNode.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { NodeBalancerConfigNode } from 'linode-js-sdk/lib/nodebalancers';
```
Comment (Jskobos) [inline packages/manager/src/store/nodeBalancerConfigNodes/configNode.reducer.ts]: This reducer should be unit-tested

**Comment by Jskobos** [comment]
Closing as out of date. Can re-open or rebuild from the commits.

**Pull request closed without merging**